src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -273,10 +274,12 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
             }
             m_compiled_submodels[id].param_base = fcn_template._param_offset;
             m_compiled_submodels[id].closure = subgraph._closure;
+            m_compiled_submodels[id].transformations = subgraph._transformations;
             m_compiled_submodels[id].scales = subgraph._scales;
             m_compiled_submodels[id].zerops = subgraph._zerops;
-            m_compiled_submodels[id].update_required.resize(subgraph._closure.size(), false);
-            fill_weights_bank(id);
+            m_compiled_submodels[id].update_required.resize(subgraph._closure.size(),
+                                                            m_cfg.get<::intel_npu::NPUW_FOLD>() ? true : false);
```

Comment:
since the value you set here doesn't depend on the `id`, you can identify your flag value just once before the block.

Reply:
Done